server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2644,6 +2643,9 @@ private void verifyVmLimits(UserVmVO vmInstance, Map<String, String> details) {
             Map<String, String> customParameters = new HashMap<>();
             customParameters.put(VmDetailConstants.CPU_NUMBER, String.valueOf(newCpu));
             customParameters.put(VmDetailConstants.MEMORY, String.valueOf(newMemory));
+            if (svcOffering.getSpeed() == null) {
```

Comment:
@shwstppr do we need similar checks for CPU number and memory?

Reply:
@harikrishna-patnala cpu and memory are already set in customParameters variable above

Reply:
@shwstppr This is about the null check, but yeah as @Pearl1594 mentioned we are checking it in another method.